### ISSUE TITLE
Repo Layer 

### ISSUE THREAD

**Issue by justinengle-krs** [stock2shop/console_v2#70]
@matthampton-stock2shop Something we do in many of our projects is add a `repo` layer that talks to the outside world

Before I go to far, I'd like to get your thoughts.

Especially with storybook and requiring us to `Mock` our API's,  our components currently have parameters where we pass the functions in.

Instead of the functions, I'd like to `inject` the repo's which contain these functions. I think it lends itself well to well defined classes and a neat project structure.

so example:

product-repo.ts
```
import { SystemProductMetaDataWrapper } from '@/models/products/product-meta-data-wrapper';
import type { SystemProductMeta } from './../models/products/product-meta';

export class ProductsRepo {
  async get(): Promise<SystemProductMeta> {
    const opts: APIRequest<undefined> = {
      path: `products/meta`
    };
    const result = await get<SystemProductMetaDataWrapper>(opts);
    return new SystemProductMetaDataWrapper(result).system_product_meta;
  }
}
```

Products.svelte

```
<script>
let productRepo = new ProductRepo()
</script>
<Product {productRepo} />
``

Products.stories.svelte

```
<script>
let productRepo = new MockProductRepo()
</script>
<Product {productRepo} />
``

**Comment by matthampton-stock2shop** [comment]
@justinengle-krs I think that looks good ... 

@flumonion Are you ok with this?

**Comment by matthampton-stock2shop** [comment]
One thing to consider is that the infrastructure for interacting with elastic search that is already there is already quite sophisticated - it uses a svelte context instead (which is also then not so hard to mock). 

But I think this Repo idea is good for wrapping the straightforward CRUD type operations

**Comment by justinengle-krs** [comment]
> One thing to consider is that the infrastructure for interacting with elastic search that is already there is already quite sophisticated - it uses a svelte context instead of a prop (which is also then not so hard to mock - I have already mocked it for products page story).
> 
> But I think this Repo idea is good for wrapping the straightforward CRUD type operations

Yes, I've seen, I guess that can stay as is for now...not a huge fan of the 'global' context for the store. But it seems quite complex, so I'd rather not touch that for now

**Comment by flumonion** [comment]
> @justinengle-krs I think that looks good ...
> 
> @flumonion Are you ok with this?

Happy to go with this suggestion.
